### PR TITLE
Configurable base twist controller

### PR DESCRIPTION
**Pull request by fmessmer**
#includes #134 

related to #91 
For the moment, we can only reconfigure the default values of the SteerCtrl parameters, i.e. the parameters are applied to all wheels...

@ipa-mdl 
Do you see any issue with this implementation regarding interference with the controller read-update-write cycle?

### CODE REVIEW

**Review by mathias-luedtke — CHANGES_REQUESTED** [review]
This cannot work properly as it is not thread-safe and not real-time-safe.

File: cob_omni_drive_controller/src/control_plugin.cpp [hunk 1]
```diff
@@ -149,7 +155,15 @@ class WheelController: public GeomController<hardware_interface::VelocityJointIn
         }
     }
 
-
+    boost::recursive_mutex reconfig_mutex_;
```
Comment (mathias-luedtke) [inline cob_omni_drive_controller/src/control_plugin.cpp]: recursive mutexes are evil.

File: cob_omni_drive_controller/src/control_plugin.cpp [hunk 2]
```diff
@@ -149,7 +155,15 @@ class WheelController: public GeomController<hardware_interface::VelocityJointIn
...
+    boost::shared_ptr< dynamic_reconfigure::Server<SteerCtrlConfig> > reconfigure_server_;
+    
+    void reconfigureCallback(SteerCtrlConfig &config, uint32_t level) {
+        geom_->reconfigureSteerCtrlParams(config);
```
Comment (mathias-luedtke) [inline cob_omni_drive_controller/src/control_plugin.cpp]: not thread safe.

store data in realtime buffer and process in in `update()`.

**Comment by mathias-luedtke** [comment]
~~please merge bf7f138 separately.~~ #134 
I will provide a fix for cab763c

**Comment by fmessmer** [comment]
Pull in @ipa-mdl's fixes + rebase as #134 has been merged

**Comment by mathias-luedtke** [comment]
I was not yet ready ;)
Per-joint option is missing, I will add (and squash) it tomorrow.

**Comment by fmessmer** [comment]
@ipa-mdl 
Current implementation so far seems to work, i.e. rqt-reconfigured parameters are considered within the base_twist_controller, e.g. `d_phi_max`:
reconfigure_dphimax

**Comment by fmessmer** [comment]
closing in favor of #137

**Pull request closed without merging**